### PR TITLE
README refresh

### PR DESCRIPTION
**Pull request by mateuszsrebrny-reef**
In this PR:
- [x] the explonatory and mission part of the README gets rewritten
- [x] the picture looses 3rd party systems (we are focusing on other subnet's validators)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,60 +1,114 @@
...
+   Add support for all GPU classes required by other Bittensor subnets.
+
+1. **Fair Resource Sharing**  
+   Allocate resources based on validators' stakes while allowing low-stake validators access when demand is low.
+
+# Running ComputeHorde components
+
+This repository contains the reference implementations of:
+
+- **Validator**: A reference implementation requiring a Trusted Miner for cross-checking synthetic tasks.
```
Comment (mpnowacki-reef) [inline README.md:105]: while this is all true in principle, I do think we would like to discourage subnet members from modifying:

1. validator
2. executor
3. miner in different regard than ExecutorManager

so I would avoid the word "reference"

Comment (mateuszsrebrny-reef) [reply README.md]: gotcha :+1: check out the fixed version pls @mpnowacki-reef : https://github.com/backend-developers-ltd/ComputeHorde/pull/340/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R101-R114

File: README.md [hunk 2]
```diff
@@ -100,15 +102,18 @@ ComputeHorde adds GPU-powered validation to this ecosystem, helping other subnet
 
 # Running ComputeHorde components
 
-This repository contains the reference implementations of:
+This repository contains the implementations of:
 
-- **Validator**: A reference implementation requiring a Trusted Miner for cross-checking synthetic tasks.
+- **Validator**: Requres a Trusted Miner for cross-checking synthetic tasks.
```
Comment (mpnowacki-reef) [inline README.md:107]:
```suggestion
- **Validator**: Requires a Trusted Miner for cross-checking synthetic tasks.
```

File: README.md [hunk 3]
```diff
@@ -100,15 +102,18 @@ ComputeHorde adds GPU-powered validation to this ecosystem, helping other subnet
...
 - **Miner**: Default miner setup with a single executor.
```
Comment (mpnowacki-reef) [inline README.md:108]: still the `Default` incentives making "your own"

File: README.md [hunk 4]
```diff
@@ -100,15 +102,18 @@ ComputeHorde adds GPU-powered validation to this ecosystem, helping other subnet
...
-- **Executor**: Base implementation for executing dockerized jobs. Users can create custom executor managers to scale and optimize mining efficiency.
+- **Executor**: Base implementation for executing dockerized jobs.
```
Comment (mpnowacki-reef) [inline README.md:109]: it's not a `Base` it's the one and only

**Comment by mateuszsrebrny-reef** [comment]
Ok @mpnowacki-reef this is the polished version of what we discussed:
- trusted miner intro: https://github.com/backend-developers-ltd/ComputeHorde/pull/340/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R59-R62

- reworked running components intro https://github.com/backend-developers-ltd/ComputeHorde/pull/340/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R111-R115

what say you?

File: README.md [hunk 5]
```diff
@@ -1,60 +1,124 @@
...
+
+# Running ComputeHorde components
+
+This repository contains the implementations of:
+
+- **Validator**: Requires a [Trusted Miner](#validator) for cross-checking synthetic tasks.
+- **Miner**: Modifying the miner code on subnet 12 is discouraged, as the stock implementation manages only communications between components.
+  The competitive edge lies in optimizing executor provisioning.
+  Users can create [custom executor managers](miner#custom-executor-manager) to scale and optimize mining efficiency.
+  The default implementation runs a single executor and is not intended for mainnet use.
```
Comment (mpnowacki-reef) [inline README.md:115]:
```suggestion
  The default executor manager runs a single executor and is not intended for mainnet use.
```

File: README.md [hunk 6]
```diff
@@ -1,60 +1,124 @@
...
-# ComputeHorde
+- **Scalable Mining with Executors**  
+  Each miner in ComputeHorde can spawn multiple **executors**, performing individual compute tasks. 
+  This removes the 256 miner (UID) limit and significantly scales the potentially available computing power.
 
-![ComputeHorde.png](ComputeHorde.png)
+- **Hardware Classes**  
+  ComputeHorde introduces hardware classes to create a free market for GPU resources, balancing cost-effectiveness with performance. 
+  Currently, **A6000** is the supported class, with **A100** coming next.
+  The end goal is to support all GPU types/configurations required by validators across Bittensor subnets.
```
Comment (ppolewicz) [inline README.md:33]:
```suggestion
  The end goal is to eventually support all GPU types/configurations required by validators across Bittensor subnets.
```

File: README.md [hunk 7]
```diff
@@ -1,60 +1,124 @@
...
-* Ensure that benchmark jobs are paid in the same manner as organic jobs: job duration multiplied by the hardware class multiplier and the benchmark value.
-* When a new miner is registered, require all validators to benchmark the miner's hardware classes with extended timeouts to accurately assess their capabilities.
-* When a new validator registers, they must benchmark every other miner in the network to maintain an up-to-date and comprehensive understanding of available resources. Until a miner is benchmarked by the validator, the validator defaults to 1 as the locally_measured_efficiency_factor for that miner.
-* Miners will have the ability to modify their hardware class availability manifest at a frequency of once every 2 hours. In the event that a miner has available executors, they are obligated to accept assigned jobs and cannot reject them. Should a miner reject a job under such circumstances, the validator will impose a penalty by lowering the hardware class local multiplier for all tasks associated with that miner.
+### **Facilitator**
+- Acts as a gateway for organic requests (from other subnets’ validators) to enter ComputeHorde.
```
Comment (ppolewicz) [inline README.md:51]: I don't think that's true anymore, check with Even

File: README.md [hunk 8]
```diff
@@ -1,60 +1,124 @@
...
+- Sends tasks to chosen validators, who then distribute them to miners.
 
+### **Validator**
+- Receives organic requests via the Facilitator or generates synthetic tasks for validation.
+- Distributes tasks to miners and evaluates the results:
+  - Organic results are returned to external requesters.
```
Comment (ppolewicz) [inline README.md:57]: this will change next week

File: README.md [hunk 9]
```diff
@@ -1,60 +1,124 @@
...
+  - Synthetic results adjust miners' scores.
+- Uses a separate GPU, called a **Trusted Miner**, to pre-run part of the validation tasks and establish expected results. 
+  The Trusted Miner shares the same code as a regular miner but is configured differently:
```
Comment (ppolewicz) [inline README.md:60]:
```suggestion
  The Trusted Miner shares the same code as a regular miner, but is configured differently:
```

File: README.md [hunk 10]
```diff
@@ -1,60 +1,124 @@
...
+## Innovations to Highlight
+
+### Discouraging Weight-Copying
+- Commit-Reveal: Validators post hidden weights and reveal them in the next epoch, making the copying of _current_ weights impossible.
+- Executor Dancing: Miners randomly move GPUs across multiple UIDs, further reducing the effectiveness of copying old weights.
+
+### Encouraging Actual Mining
+- Synthetic tasks are designed to run only on specific hardware (e.g., A6000 GPUs), ensuring miners deliver the advertised compute power.
+- Scoring system incentivizing for completing organic tasks.
+- Mechanisms for penalizing miners serving only a subset of validators
```
Comment (ppolewicz) [inline README.md:88]:
```suggestion
```
I don't think we have one yet

File: README.md [hunk 11]
```diff
@@ -1,60 +1,124 @@
...
 
-of the ComputeHorde BitTensor SubNet. Running etc. is explained in each component's README.
+- **Decentralized Compute for Bittensor Validators**  
+  ComputeHorde aims to become the go-to decentralized source for hardware needed to validate other subnets.
+  The mission is to decrease the Bittensor ecosystem's dependency on centralized services.
+  The assurance of the miners' work quality is essential for the Bittensor's overall reliability.
+  
+- **Fair and Verified Work**  
+  ComputeHorde employs mechanisms to ensure miners provide authentic compute work, fairly verified by the validators:
+  - Execute tasks from all validators equally, regardless of their stake (above a rather low threshold).
```
[the remark this answers is not in the document]
Comment (mateuszsrebrny-reef) [reply README.md:21]: that is what I meant - `not ignoring` equally, regardless of stake, but still proportionally ;)
but I agree, your version is much clear :+1:

File: README.md (hunk 7 above)
Comment (mateuszsrebrny-reef) [reply README.md:51]: It is imo, the external requesters aka other subnets' valis push tasks (organic tasks) to **their** facilitator and it then pushes it to **their** sn12 valis - stock version pushes to the one vali, but it is imaginable that it has more whitelisted.
Will confirm and rethink the wording :+1:

File: README.md (hunk 8 above)
Comment (mateuszsrebrny-reef) [reply README.md:57]: I think the same as here https://github.com/backend-developers-ltd/ComputeHorde/pull/340/files#r1886261661
Will confirm :)

File: README.md (hunk 7 above)
Comment (mateuszsrebrny-reef) [reply README.md:51]: decided to push it, without Even, will ask him and potentially fix it when he is available

**Pull request merged**